File: Dataset/Leetcode/test/27/432.java
 class Solution {
    public int XXX(int[] nums, int val) {
        int len = nums.length;
        if(nums == null || len == 0){
            return 0;
        }
        int count = 0;
        for(int i= 0;i< len;i++){
            if(nums[i] == val){
                count++;
            }
        }
        int i = 0;
        int j = len-1;
        while(i < j){
            if(nums[j] == val){
                j--;
            }
            if(nums[i] != val){
                i++;
            }else if(nums[i]==val && nums[j] != val){
                swap(nums,i,j);
                i++;
                j--;
            }
        }
        return len-count;
    }

    public void swap(int[] nums,int i,int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: Dataset/Leetcode/test/78/432.cpp
 class Solution {
public:
    vector<vector<int>> XXX(vector<int>& nums) {
        _result.clear();
        _subSet.clear();
        generateResult(nums, 0);
        return _result;
    }

    void generateResult(vector<int>& nums, size_t start) {
        _result.push_back(_subSet);
        for (size_t loop = start; loop < nums.size(); ++loop) {
            _subSet.push_back(nums[loop]);
            generateResult(nums, loop + 1);
            _subSet.pop_back();
        }
    }
private:
    vector<vector<int>> _result;
    vector<int> _subSet;
};

